src/psyclone/tests/dynamo0p3_builtins_test.py

```diff
@@ -2570,6 +2617,9 @@ def test_multi_builtin_single_invoke():  # pylint: disable=invalid-name
 # ------------- Invalid built-in with an integer scalar reduction ----------- #
 
 
+@pytest.mark.xfail(
+    reason="Reduction into an integer variable is not currently supported "
+    "by the LFRic infrastructure.")
 def test_scalar_int_builtin_error(monkeypatch):
```

Comment:
I don't understand why this test is now marked as xfail. The purpose of the test is to check that PSyclone rejects an attempt to perform a reduction into an integer variable. This is still forbidden and therefore we still want to check that PSyclone rejects it with an appropriate error message.

Reply:
I removed the xfail mark and tried to deal with the Generation Error as I saw in other tests. The test fails and I am not really sure what I did wrong. Help would be much appreciated, thank you!

Reply:
SOLVED: As the default is gfortran, all it needed was to explicitly specify --f90="compiler name".

py.test --compile --f90="pgfortran" src/psyclone/tests/dynamo0p3_builtins_test.py for Portland

py.test --compile --f90="ifort" src/psyclone/tests/dynamo0p3_builtins_test.py for Intel

It is a bit slower than Gnu but it works :)